dev_scripts/importTranslation.js

```diff
@@ -0,0 +1,111 @@
+/* eslint-disable default-case */
+/* eslint-disable prettier/prettier */
+/* eslint-disable no-useless-escape */
+/* eslint-disable array-callback-return */
+/* eslint-disable import/no-dynamic-require */
+/* eslint-disable global-require */
+/* eslint-disable no-return-assign */
+/* eslint-disable no-shadow */
+/* eslint-disable guard-for-in */
+/* eslint-disable prefer-destructuring */
+/* eslint-disable no-undef */
+/* eslint-disable no-restricted-syntax */
+
+/**
+ * mSupply Mobile
+ * Sustainable Solutions (NZ) Ltd. 2019
+ */
+
+const fs = require('fs');
+
+// To identify {nextLine} in the csv files
+const nextLine = /{nextLine}/gi;
+// To identify {emptySpace} in the csv files
+const emptySpace = /{emptySpace}/gi;
+const localizationFiles = [];
+const keys = [];
+const values = [];
+let localizationFileName = [];
+let key = null;
+let value = null;
+let index = 0;
+
+getLocalizationFiles = () => {
+  // Reads all files in localization files and save them in localizationFiles array
+  fs.readdirSync('././src/localization/').forEach(file => {
+    // Filter to only use get JSON files
+    if (file.split('.').pop() === 'json') {
+      const fileName = require(`../src/localization/${file}`);
+      localizationFiles.push(fileName);
+      localizationFiles[index].name = file.replace('.json', '');
+      index += 1;
+    }
+  });
+};
+// Gets the language entered by the User (fr, la, gil, tl)
+getSelectedLanguage = () => process.argv[2];
+
+UpdateJsonFiles = selectedLanguage => {
+  // Reads the updated csv file chosen by the user (fr.csv, la.csv, etc)
+  const fileContent = fs.readFileSync(
+    `./src/localization/translations/${selectedLanguage}.csv`,
+    'utf8'
+  );
+  // Creates one string per line available in the csv file
+  const arrayOfData = fileContent.toString().split('\n');
+
+  // Go through all localizationFiles
+  localizationFiles.map(localizationFile => {
+    arrayOfData.map((arrayOfData, indexOfArrayOfData) => {
+      const data = JSON.stringify(arrayOfData)
+        .replace(/"/g, '')
+        .split(',');
+      // Gets [ key, Value in mainLanguage, Value in selectedLanguage] data
+      data[indexOfArrayOfData] = arrayOfData.split(',').slice(1);
```

Comment:
as per our discussion, if you log the following, you'll see result:
![Screen Shot 2019-12-16 at 2 19 04 PM](https://user-images.githubusercontent.com/26194949/70872749-13fa7200-200f-11ea-8931-7abaeea7e6f5.png)
<img width="446" alt="Screen Shot 2019-12-16 at 2 18 56 PM" src="https://user-images.githubusercontent.com/26194949/70872754-18bf2600-200f-11ea-8905-67729c5a8378.png">

Looks like you are creating an array inside of array. 

Anyways, my suggestion for the logic in import:
1 - parse all of arrayOfData into structure like this:
```
{
 ${fileName}: {
    ${stringKey}: {$stringValue}
  ...
 }
 ...
}
```
Btw it's nice to use constants for positions in the row, i.e. csvRow[fileName], csvRow[translationKey] etc..

2 - Iterate through localizationFiles and make key/value object, checking if general translation key exists with a value in the object you've created above.
3 - Instead of `localizationFile[selectedLanguage] = resultObject;` use Object.assign (destructuring would be better, but we don't have ES functionality in these scripts)

If you want, we can do 30 mins of peer programming and I can show what I mean : )

Reply:
Thank you so much for this one, I could reduce so many lines following your advices. Highly appreciated!